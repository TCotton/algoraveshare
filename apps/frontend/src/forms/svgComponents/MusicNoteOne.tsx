import * as React from 'react'
import type { SVGProps } from 'react'

const MusicNoteOne = React.forwardRef<SVGSVGElement, SVGProps<SVGSVGElement>>(
  (props, ref) => (
    <svg
      ref={ref}
      xmlns='http://www.w3.org/2000/svg'
      viewBox='0,0,256,256'
      width='144px'
      height='144px'
      fillRule='nonzero'
      className='svg-icon'
      {...props}
    >
      <g
        fill='none'
        fillRule='nonzero'
        stroke='#b4befe'
        strokeWidth='3'
        strokeLinecap='round'
        strokeLinejoin='round'
        strokeMiterlimit='10'
        strokeDasharray=''
        strokeDashoffset='0'
        fontFamily='none'
        fontWeight='none'
        fontSize='none'
      >
        <g transform='scale(5.33333,5.33333)'>
          <path className='path-one' data-animation='stop' d='M24.5,17.5v13.4'></path>
          <path
            className='path-two'
            data-animation='stop'
            d='M17.8,28.5c0.7,-0.3 1.5,-0.5 2.4,-0.5c2.4,0 4.4,1.5 4.4,3.3c0,1.8 -2,3.3 -4.4,3.3c-1.6,0 -3,-0.6 -3.7,-1.6'
          >
          </path>
          <path
            className='path-three'
            data-animation='stop'
            d='M29.5,20.6l2.4,0.6c0.6,0.2 1.3,-0.3 1.3,-1v-2.8c0,-1 -0.7,-1.9 -1.7,-2.1l-5.8,-1.6c-0.6,-0.2 -1.3,0.3 -1.3,1v4.5'
          >
          </path>
          <path
            className='path-four'
            data-animation='stop'
            d='M31,41.1c-2.2,0.9 -4.5,1.4 -7,1.4c-10.2,0 -18.5,-8.3 -18.5,-18.5c0,-3.9 1.2,-7.6 3.3,-10.6'
          >
          </path>
          <path
            className='path-five'
            data-animation='stop'
            d='M15.8,7.4c2.5,-1.2 5.2,-1.9 8.2,-1.9c10.2,0 18.5,8.3 18.5,18.5c0,4.8 -1.8,9.2 -4.8,12.5'
          >
          </path>
        </g>
      </g>
    </svg>
  )
)

MusicNoteOne.displayName = 'MusicNoteOne'

export default MusicNoteOne
