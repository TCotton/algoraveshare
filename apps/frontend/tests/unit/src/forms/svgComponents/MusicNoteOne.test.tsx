import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import MusicNoteOne from '../../../../../src/forms/svgComponents/MusicNoteOne'

describe('MusicNoteOne', () => {
  it('renders svg element with correct attributes', () => {
    const { container } = render(<MusicNoteOne />)

    const svg = container.querySelector('svg')
    expect(svg).toBeInTheDocument()
    expect(svg).toHaveAttribute('xmlns', 'http://www.w3.org/2000/svg')
    expect(svg).toHaveAttribute('viewBox', '0,0,256,256')
    expect(svg).toHaveAttribute('width', '144px')
    expect(svg).toHaveAttribute('height', '144px')
    expect(svg).toHaveAttribute('fill-rule', 'nonzero')
    expect(svg).toHaveClass('svg-icon')
  })

  it('renders all five path elements with correct classes', () => {
    const { container } = render(<MusicNoteOne />)

    const pathOne = container.querySelector('.path-one')
    const pathTwo = container.querySelector('.path-two')
    const pathThree = container.querySelector('.path-three')
    const pathFour = container.querySelector('.path-four')
    const pathFive = container.querySelector('.path-five')

    expect(pathOne).toBeInTheDocument()
    expect(pathTwo).toBeInTheDocument()
    expect(pathThree).toBeInTheDocument()
    expect(pathFour).toBeInTheDocument()
    expect(pathFive).toBeInTheDocument()
  })

  it('all paths have data-animation attribute set to stop', () => {
    const { container } = render(<MusicNoteOne />)

    const paths = container.querySelectorAll('path')
    expect(paths).toHaveLength(5)

    paths.forEach((path) => {
      expect(path).toHaveAttribute('data-animation', 'stop')
    })
  })

  it('forwards ref correctly', () => {
    const ref = React.createRef<SVGSVGElement>()

    render(<MusicNoteOne ref={ref} />)

    expect(ref.current).toBeInstanceOf(SVGSVGElement)
    expect(ref.current?.tagName).toBe('svg')
  })

  it('accepts additional SVG props via rest props', () => {
    const { container } = render(
      <MusicNoteOne
        data-testid='custom-svg'
        aria-label='Music note icon'
        id='custom-id'
        style={{ opacity: 0.5 }}
      />
    )

    const svg = container.querySelector('svg')
    expect(svg).toHaveAttribute('data-testid', 'custom-svg')
    expect(svg).toHaveAttribute('aria-label', 'Music note icon')
    expect(svg).toHaveAttribute('id', 'custom-id')
    expect(svg).toHaveStyle({ opacity: '0.5' })
  })

  it('allows custom className to override default class', () => {
    const { container } = render(<MusicNoteOne className='custom-class' />)

    const svg = container.querySelector('svg')
    expect(svg).toHaveClass('custom-class')
    expect(svg).not.toHaveClass('svg-icon')
  })

  it('supports imperative DOM access via ref', () => {
    const ref = React.createRef<SVGSVGElement>()

    render(<MusicNoteOne ref={ref} />)

    // Test that we can query paths through the ref
    const paths = ref.current?.querySelectorAll('path')
    expect(paths).toHaveLength(5)
  })

  it('allows style manipulation through ref', () => {
    const ref = React.createRef<SVGSVGElement>()

    render(<MusicNoteOne ref={ref} />)

    // Test that we can manipulate animation-play-state on paths
    const paths = ref.current?.querySelectorAll('path')
    paths?.forEach((path) => {
      const svgPath = path as SVGPathElement
      svgPath.style.animationPlayState = 'running'
      expect(svgPath.style.animationPlayState).toBe('running')
    })
  })

  it('has correct displayName', () => {
    expect(MusicNoteOne.displayName).toBe('MusicNoteOne')
  })

  it('renders g element with correct stroke attributes', () => {
    const { container } = render(<MusicNoteOne />)

    const g = container.querySelector('g[fill="none"]')
    expect(g).toBeInTheDocument()
    expect(g).toHaveAttribute('stroke', '#b4befe')
    expect(g).toHaveAttribute('stroke-width', '3')
    expect(g).toHaveAttribute('stroke-linecap', 'round')
    expect(g).toHaveAttribute('stroke-linejoin', 'round')
  })

  it('matches snapshot', () => {
    const { container } = render(<MusicNoteOne />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
