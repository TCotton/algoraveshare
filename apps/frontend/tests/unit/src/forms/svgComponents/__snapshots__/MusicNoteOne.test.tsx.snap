// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MusicNoteOne > matches snapshot 1`] = `
<svg
  class="svg-icon"
  fill-rule="nonzero"
  height="144px"
  viewBox="0,0,256,256"
  width="144px"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fill-rule="nonzero"
    font-family="none"
    font-size="none"
    font-weight="none"
    stroke="#b4befe"
    stroke-dasharray=""
    stroke-dashoffset="0"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-miterlimit="10"
    stroke-width="3"
  >
    <g
      transform="scale(5.33333,5.33333)"
    >
      <path
        class="path-one"
        d="M24.5,17.5v13.4"
        data-animation="stop"
      />
      <path
        class="path-two"
        d="M17.8,28.5c0.7,-0.3 1.5,-0.5 2.4,-0.5c2.4,0 4.4,1.5 4.4,3.3c0,1.8 -2,3.3 -4.4,3.3c-1.6,0 -3,-0.6 -3.7,-1.6"
        data-animation="stop"
      />
      <path
        class="path-three"
        d="M29.5,20.6l2.4,0.6c0.6,0.2 1.3,-0.3 1.3,-1v-2.8c0,-1 -0.7,-1.9 -1.7,-2.1l-5.8,-1.6c-0.6,-0.2 -1.3,0.3 -1.3,1v4.5"
        data-animation="stop"
      />
      <path
        class="path-four"
        d="M31,41.1c-2.2,0.9 -4.5,1.4 -7,1.4c-10.2,0 -18.5,-8.3 -18.5,-18.5c0,-3.9 1.2,-7.6 3.3,-10.6"
        data-animation="stop"
      />
      <path
        class="path-five"
        d="M15.8,7.4c2.5,-1.2 5.2,-1.9 8.2,-1.9c10.2,0 18.5,8.3 18.5,18.5c0,4.8 -1.8,9.2 -4.8,12.5"
        data-animation="stop"
      />
    </g>
  </g>
</svg>
`;
